src/connections/connections.controller.ts

```diff
@@ -0,0 +1,97 @@
+import { Controller, Delete, Get, Param, Post } from '@nestjs/common';
+import { CurrentUser } from 'src/decorators/current-user.decorator';
+import { ConnectionsService } from './connections.service';
+import { ConnectionDto } from './DTO/Connection.dto';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiNotFoundResponse,
+} from '@nestjs/swagger';
+import { User } from '@prisma/client';
+import { Public } from 'src/decorators/public.decorator';
+
+@ApiBearerAuth()
+@Controller('connections')
+export class ConnectionsController {
+  constructor(private connectionsService: ConnectionsService) {}
+  /**
+   *
+   * Get all of the current user's connections
+   */
+  @Get()
+  async getAllConnections(@CurrentUser() user: User): Promise<ConnectionDto[]> {
+    return this.connectionsService.getUserConnections(user.id);
+  }
+
+  /**
+   * Get "suggested for review" connections.
+   */
+  @Get('/suggested')
+  async getSuggestedConnections(
+    @CurrentUser() user: User,
+  ): Promise<ConnectionDto[]> {
+    return this.connectionsService.getUserConnections(user.id);
+  }
+
+  /**
+   * Search an user by query.
+   */
+  @Get('/search/:query')
+  @Public()
+  @ApiBadRequestResponse()
+  async searchUser(
+    @Param('query') query: string,
```

Comment:
Maybe you can add an alphanumeric regex check in here so that users cant use SQL injection?

Reply:
If prisma wouldn't be Sql injection safe out of the box all of our endpoint would be vulnerable for such a thing since we're not doing any kind of prevention for this anywhere. 
Glad that is safe out of the box >.< 
https://stackoverflow.com/questions/73943274/prisma-findunique-is-sql-injection-safe